articles/quickstart/spa/auth0-react/01.md

```diff
@@ -0,0 +1,299 @@
+
+- `isLoading`: Returns `false` when the SDK has created an Auth0 Client and has determined the user's authentication status.
+
+- `login()`: Takes your users to the [Auth0 Universal Login](https://auth0.com/docs/universal-login) page and back to your application once they log in.
+
+- `logout()`: Logs out the user from your application.
+
+```javascript
+import React from "react";
+import "./App.css";
+import { useAuth0 } from "@auth0/auth0-react";
+
+function App() {
+  const { isAuthenticated, isLoading, login, logout } = useAuth0();
+
+  return (
+    <div className="App">
+      {isLoading ? (
+        <h2>Loading...</h2>
+      ) : (
+        <>
+          {isAuthenticated ? (
+            <button onClick={() => logout()}>Log Out</button>
+          ) : (
+            <button onClick={() => login()}>Log In</button>
+          )}
+        </>
+      )}
+    </div>
+  );
+}
+
+export default App;
+```
+
+The `App` component above displays a `Loading...` message while the SDK completes its initialization process: creating an Auth0 Client and determining the user's authentication status. Once the initialization is complete, the component renders a "Log In" button if Auth0 has not authenticated the user yet; otherwise, it renders a "Log Out" button.
+
+> By default, Auth0 offers you developer keys to test logging in with a [social identity provider](https://auth0.com/docs/connections/identity-providers-social), such as Google. These developer keys have some limitations that may cause your app to behave differently. For more details on this and on how to fix it, read on ["Test Social Connections with Auth0 Developer Keys"](https://auth0.com/docs/connections/social/devkeys#limitations-of-developer-keys).
+
+## Read the User Profile
+
+Once you have users stored in your Auth0 user database, the Auth0 React SDK helps you retrieve the profile information associated with logged-in users quickly in whatever component you need it, such as their name or profile picture to personalize a page. The profile information is available through the `user` object exposed by the `useAuth0()` custom React Hook.
+
+Take for example this `Profile` component:
+
+```javascript
+import React from "react";
+import { useAuth0 } from "@auth0/auth0-react";
+
+const Profile = () => {
+  const { user } = useAuth0();
+
+  return (
+    <div>
+      <img src={user.picture} alt={user.name} />
+      <h2>{user.name}</h2>
+      <p>{user.email}</p>
+    </div>
+  );
+};
+
+export default Profile;
+```
+
+The `user` property from `useAuth()` contains sensitive information and artifacts related to the user's identity. As such, you can prevent leaking sensitive information to the public by checking if your app has initialized the SDK correctly and if Auth0 has authenticated the user before you render any component that consumes the `user` property. An alternative to doing manual checks within a component is to prevent the whole component from rendering using routing.
```

Comment:
"As such, you can prevent leaking sensitive information to the public by checking if your app has initialized the SDK correctly" 

Don't think this is correct, you wouldn't leak that information if you didn't check the user was authenticated before visiting it, you'd just see a blank page.

Reply:
Moved to CF.